idaes/models/unit_models/tests/test_gibbs_scaling.py

```diff
@@ -0,0 +1,464 @@
+        )
+        set_scaling_factor(
+            model.fs.unit.control_volume.properties_out[0.0].mole_frac_comp["CH4"], 1e4
+        )
+        set_scaling_factor(
+            model.fs.unit.control_volume.properties_out[0.0].mole_frac_comp["CO"], 1e3
+        )
+        set_scaling_factor(
+            model.fs.unit.control_volume.properties_out[0.0].mole_frac_comp["H2O"], 10
+        )
+        set_scaling_factor(
+            model.fs.unit.control_volume.properties_out[0.0].mole_frac_comp["NH3"], 1e4
+        )
+        set_scaling_factor(
+            model.fs.unit.control_volume.properties_out[0.0].temperature, 1e-3
+        )
+        set_scaling_factor(
+            model.fs.unit.control_volume.properties_out[0.0].pressure, 1e-5
+        )
+
+        from_json(model, fname=fname, wts=StoreSpec.value())
+
+        return model
+
+    def test_variable_scaling_only(self, methane):
+        unscaled = jacobian_cond(methane, scaled=False)
+
+        scaler = GibbsReactorScaler()
+        scaler.variable_scaling_routine(
+            methane.fs.unit,
+            submodel_scalers={
+                "control_volume.properties_in": SMScaler,
+                "control_volume.properties_out": SMScaler,
+            },
+        )
+
+        scaled = jacobian_cond(methane, scaled=True)
+
+        count = 0
+        for c in methane.component_data_objects(ctype=[Var], descend_into=True):
+            sf = get_scaling_factor(c)
+            if sf is None:
+                count += 1
+        assert count == 52
+
+        assert scaled < unscaled
+        assert scaled == pytest.approx(8.908989e16, rel=1e-5)
+
+    def test_constraint_scaling_only(self, methane):
+        unscaled = jacobian_cond(methane, scaled=False)
+
+        scaler = GibbsReactorScaler()
+        scaler.constraint_scaling_routine(
+            methane.fs.unit,
+            submodel_scalers={
+                "control_volume.properties_in": SMScaler,
+                "control_volume.properties_out": SMScaler,
+            },
+        )
+
+        scaled = jacobian_cond(methane, scaled=True)
+
+        count = 0
+        for c in methane.component_data_objects(ctype=[Constraint], descend_into=True):
+            sf = get_scaling_factor(c)
+            if sf is None:
+                count += 1
+        assert count == 0
+
+        assert scaled < unscaled
+        assert scaled == pytest.approx(9.093003e15, rel=1e-5)
+
+    def test_full_scaling(self, methane):
+        unscaled = jacobian_cond(methane, scaled=False)
+
+        scaler = GibbsReactorScaler()
+        scaler.scale_model(
+            methane.fs.unit,
+            submodel_scalers={
+                "control_volume.properties_in": SMScaler,
+                "control_volume.properties_out": SMScaler,
+            },
+        )
+
+        scaled = jacobian_cond(methane, scaled=True)
+
+        assert scaled < unscaled
+        assert scaled == pytest.approx(6.96238e15, rel=1e-5)
```

Comment:
Once this gets merged, I want to see what's causing the condition number to still be so badly behaved.

Reply:
My guess is the fact that the model is uninitialized, but I gave it good initial guesses for the actual final state - i.e. there is a mismatch between the current uninitialized state and the scaling factors it was given. If we went and solved the model, I suspect the value would be much lower.